apps/server/src/modules/learnroom/mapper/board.mapper.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { courseFactory, taskFactory, setupEntities } from '@shared/testing';
+import { BoardMapper } from './board.mapper';
+import { BoardResponse, BoardElementResponse } from '../controller/dto/roomBoardResponse';
+
+describe('board mapper', () => {
+	let mapper: BoardMapper;
+
+	beforeAll(async () => {
+		await setupEntities();
+	});
+
+	beforeEach(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			imports: [],
+			providers: [BoardMapper],
+		}).compile();
+
+		mapper = module.get(BoardMapper);
+	});
+	describe('mapToResponse', () => {
+		it('should map plain board into response', () => {
+			const course = courseFactory.buildWithId();
+			const board = {
+				roomId: 'roomId',
+				displayColor: '#ACACAC',
+				title: 'boardTitle',
+				courseName: course.name,
+				elements: [],
+			};
+
+			const result = mapper.mapToResponse(board);
+
+			expect(result instanceof BoardResponse).toEqual(true);
+		});
+
+		it('should map tasks with status on board to response', () => {
+			const course = courseFactory.buildWithId();
+			const task = taskFactory.buildWithId({ course });
+			const status = {
+				graded: 0,
+				maxSubmissions: 0,
+				submitted: 0,
+				isDraft: false,
+				isSubstitutionTeacher: false,
+			};
+			const board = {
+				roomId: 'roomId',
+				displayColor: '#ACACAC',
+				title: 'boardTitle',
+				courseName: course.name,
```

Comment:
somehow this line seems not to be enough for the coverage tests for board.mapper.ts#L22

Reply:
yes, he wants a test for the case that course is undefined (which is not possible in our special case, but typescript doesnt know that.